packages/emitter-framework/src/python/components/class-declaration/class-declaration.tsx

```diff
@@ -0,0 +1,387 @@
+import { abcModule, typingModule } from "#python/builtins.js";
+import { type Children, code, createContentSlot, For, List, mapJoin, Show } from "@alloy-js/core";
+import * as py from "@alloy-js/python";
+import {
+  isTemplateDeclaration,
+  isTemplateDeclarationOrInstance,
+  type Interface,
+  type Model,
+  type ModelProperty,
+  type Operation,
+} from "@typespec/compiler";
+import type { TemplateDeclarationNode } from "@typespec/compiler/ast";
+import type { Typekit } from "@typespec/compiler/typekit";
+import { createRekeyableMap } from "@typespec/compiler/utils";
+import { useTsp } from "../../../core/context/tsp-context.js";
+import { reportDiagnostic } from "../../../lib.js";
+import { createDocElement } from "../../utils/doc.jsx";
+import { declarationRefkeys, efRefkey } from "../../utils/refkey.js";
+import { TypeExpression } from "../type-expression/type-expression.jsx";
+import { ClassMember } from "./class-member.jsx";
+import { MethodProvider } from "./class-method.jsx";
+
+export interface ClassDeclarationPropsWithType extends Omit<py.ClassDeclarationProps, "name"> {
+  type: Model | Interface;
+  name?: string;
+  abstract?: boolean; // Global override for the abstract flag
+  methodType?: "method" | "class" | "static"; // Global override for the method type
+}
+
+export type ClassDeclarationProps = ClassDeclarationPropsWithType | py.ClassDeclarationProps;
+
+function isTypedClassDeclarationProps(
+  props: ClassDeclarationProps,
+): props is ClassDeclarationPropsWithType {
+  return "type" in props;
+}
+
+/**
+ * Gets type members (properties or operations) from a Model or Interface.
+ * @param $ - The Typekit.
+ * @param type - The model or interface type.
+ * @returns Array of model properties or operations.
+ */
+function getTypeMembers($: Typekit, type: Model | Interface): (ModelProperty | Operation)[] {
+  if ($.model.is(type)) {
+    // For models, extract properties to render as dataclass fields
+    return Array.from($.model.getProperties(type).values());
+  } else if (type.kind === "Interface") {
+    // For interfaces, extract operations to render as abstract methods
+    return Array.from(createRekeyableMap(type.operations).values());
```

Comment:
Will `type.operations` capture the operations that are present through templating or inheritance?